shell/common/shell_test.h

```diff
@@ -31,22 +31,31 @@ namespace testing {
 
 class ShellTest : public FixtureTest {
  public:
+  struct Config {
+    // Required.
+    const Settings& settings;
+    // Defaults to &GetTaskRunnersForFixture().
+    const TaskRunners* task_runners;
```

Comment:
Please don't use a raw pointer here.

Just keep this as a value - avoids one more semantic change in the way these tests work, and avoids ambiguity around whether this can be nullptr and what that means.

Reply:
Also in general here: instead of specifying defaults in a comment, initialize the values here with a default.